components/brave_new_tab/new_tab_page.mojom

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module brave_new_tab.mojom;
+
+struct BraveBackground {
+  string author;
+  string image_url;
+  string link;
+};
+
+enum SelectedBackgroundType {
+  kBrave,
+  kCustom,
+  kSolid,
+  kGradient,
+};
+
+struct SelectedBackground {
+  SelectedBackgroundType type;
+  string value;
+};
+
+struct SponsoredImageLogo {
+  string alt;
+  string destination_url;
+  string image_url;
+};
+
+struct SponsoredImageBackground {
+  string wallpaper_id;
+  string creative_instance_id;
+  string campaign_id;
+  string image_url;
+  SponsoredImageLogo? logo;
+};
+
+// WebUI-side handler for notifications from the browser.
+interface NewTabPage {
+
+  // Called when a background-related profile preference has been updated.
+  OnBackgroundPrefsUpdated();
```

Comment:
You're right - this can be optimized. The easiest next step would be to provide the pref name (as an enum value perhaps?) and let the front end decide what to update. I think we should leave this optimization for later, since it will take some thinking through (and it's hard to tell whether it's worth it).

Reply:
Okay, happy to leave this for now. One thing to keep in mind is that it does somewhat simplify the mojom interface, as you no longer need to provide getters for any of the values (you just fire the change listener when its added). I've found the pattern makes it a bit easier to reason about where stuff comes from.

If you didn't want to add a separate listener for each different channel, you could just add all the fields into the callback

`OnBackgroundPrefsUpdates(Thing1 thing1, Thing2 thing2, ..., ThingN thingN)`

and you can simplify some of your initialization logic (instead, we just add ourselves as a listener and are notified of the current state + any changes that happen later).

Reply:
I see that we've been moving some mojo interfaces toward a push-only API. I think that the most flexible setup is a hybrid where data is pulled and update notifications are pushed. It does require more round-trips and possibly more interface methods, however. I'll keep this in mind as we move through the project - the hybrid approach is not yet set-in-stone.